daemon/system-info.sh

```diff
@@ -302,65 +302,76 @@ DISK_SIZE="unknown"
-                           (echo "${dev_major_whitelist}" | grep -q ":$(cut -f 1 -d ':' "${disk}/dev"):") && \
-                           grep -qv 1 "${disk}/removable"
-                        then
-                            size="$(($(cat "${disk}/size") * 512))"
-                            DISK_SIZE="$((DISK_SIZE + size))"
-                        fi
-                done
-        elif df --version 2>/dev/null | grep -qF "GNU coreutils" ; then
-                DISK_DETECTION="df"
-                DISK_SIZE=$(($(df -x tmpfs -x devtmpfs -x squashfs -l -B1 --output=source,size | tail -n +2 | sort -u -k 1 | awk '{print $2}' | tr '\n' '+' | head -c -1)))
-        else
-                DISK_DETECTION="df"
-                include_fs_types="ext*|btrfs|xfs|jfs|reiser*|zfs"
-                DISK_SIZE=$(($(df -T -P | tail -n +2 | sort -u -k 1 | grep "${include_fs_types}" | awk '{print $3}' | tr '\n' '+' | head -c -1) * 1024))
-        fi
+  if [ -d /sys/block ] && [ -r /proc/devices ]; then
+    dev_major_whitelist=''
+
+    # This is a list of device names used for block storage devices.
+    # These translate to the prefixs of files in `/dev` indicating the device type.
+    # They are sorted by lowest used device major number, with dynamically assigned ones at the end.
+    # We use this to look up device major numbers in `/proc/devices`
+    device_names='hd sd mfm ad ftl pd nftl dasd intfl mmcblk ub xvd rfd vbd nvme'
+
+    for name in ${device_names}; do
+      if grep -qE " ${name}\$" /proc/devices; then
+        dev_major_whitelist="${dev_major_whitelist}:$(grep -E "${name}\$" /proc/devices | sed -e 's/^[[:space:]]*//' | cut -f 1 -d ' ' | tr '\n' ':'):"
+      fi
+    done
+
+    DISK_DETECTION="sysfs"
+    DISK_SIZE="0"
+    for disk in /sys/block/*; do
+      if [ -r "${disk}/size" ] \
+        && (echo "${dev_major_whitelist}" | grep -q ":$(cut -f 1 -d ':' "${disk}/dev"):") \
+        && grep -qv 1 "${disk}/removable"; then
+        size="$(($(cat "${disk}/size") * 512))"
+        DISK_SIZE="$((DISK_SIZE + size))"
+      fi
+    done
+  elif df --version 2> /dev/null | grep -qF "GNU coreutils"; then
+    DISK_DETECTION="df"
+    DISK_SIZE=$(($(df -x tmpfs -x devtmpfs -x squashfs -l -B1 --output=source,size | tail -n +2 | sort -u -k 1 | awk '{print $2}' | tr '\n' '+' | head -c -1)))
+  else
+    DISK_DETECTION="df"
+    include_fs_types="ext*|btrfs|xfs|jfs|reiser*|zfs"
+    DISK_SIZE=$(($(df -T -P | tail -n +2 | sort -u -k 1 | grep "${include_fs_types}" | awk '{print $3}' | tr '\n' '+' | head -c -1) * 1024))
+  fi
 fi
 
+# -------------------------------------------------------------------------------------------------
+# Detect whether the node is kubernetes node
+
+HOST_IS_K8S_NODE="false"
+
+if [ -n "${KUBERNETES_SERVICE_HOST}" ] && [ -n "${KUBERNETES_SERVICE_PORT}" ]; then
+  # These env vars are set for every container managed by k8s.
+  HOST_IS_K8S_NODE="true"
+elif pgrep "kubelet"; then
```

Comment:
This condition is true if i have k8s running in a container - minikube.

Minikube container is running - `_is_k8s_node: true`
Minikube container is stopped - `_is_k8s_node: false`

Perhaps we need to exclude Minikube (local testing) setups.